docs/sql-performance-tuning.md

```diff
@@ -220,13 +220,15 @@ Coalesce hints allows the Spark SQL users to control the number of output files
 tuning and reducing the number of output files. The "COALESCE" hint only has a partition number as a
 parameter. The "REPARTITION" hint has a partition number, columns, or both of them as parameters.
 The "REPARTITION_BY_RANGE" hint must have column names and a partition number is optional.
+The "COALESCE_PARTITION" hint do not has any parameter. It can merge output files by AQE.
```

Comment:
We can have a bit more discussion about the API:
1. how about other names like `REPARTITION_BEFORE_INSERT`?
2. shall we introduce hint for the INSERT statement? It's a bit weird to have hint for write in the SELECT statement.

cc @viirya @maropu @hvanhovell